rust/processor/src/db/common/models/default_models/parquet_transactions.rs

```diff
@@ -0,0 +1,381 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unused_unit)]
+
+use super::{
+    block_metadata_transactions::BlockMetadataTransaction,
+    parquet_write_set_changes::{WriteSetChangeDetail, WriteSetChangeModel},
+};
+use crate::{
+    parquet_processors::generic_parquet_processor::{HasVersion, NamedTable},
+    utils::{
+        counters::PROCESSOR_UNKNOWN_TYPE_COUNT,
+        util::{get_clean_payload, get_clean_writeset, get_payload_type, standardize_address},
+    },
+};
+use ahash::AHashMap;
+use allocative_derive::Allocative;
+use aptos_protos::transaction::v1::{
+    transaction::{TransactionType, TxnData},
+    Transaction as TransactionPB, TransactionInfo,
+};
+use field_count::FieldCount;
+use parquet_derive::ParquetRecordWriter;
+use serde::{Deserialize, Serialize};
+
+#[derive(Allocative, Clone, Debug, Deserialize, FieldCount, Serialize, ParquetRecordWriter)]
+pub struct Transaction {
+    pub txn_version: i64,
+    pub block_height: i64,
+    pub epoch: i64,
+    pub txn_type: String,
+    pub payload: Option<String>,
+    pub payload_type: Option<String>,
+    pub gas_used: u64,
+    pub success: bool,
+    pub vm_status: String,
+    pub num_events: i64,
+    pub num_write_set_changes: i64,
+    pub txn_hash: String,
+    pub state_change_hash: String,
+    pub event_root_hash: String,
+    pub state_checkpoint_hash: Option<String>,
+    pub accumulator_root_hash: String,
+    #[allocative(skip)]
+    pub block_timestamp: chrono::NaiveDateTime,
+}
+
+impl NamedTable for Transaction {
+    const TABLE_NAME: &'static str = "transactions";
+}
+
+impl HasVersion for Transaction {
+    fn version(&self) -> i64 {
+        self.txn_version
+    }
+}
+
+impl Default for Transaction {
```

Comment:
How come we have to implement Default? There is no "default transaction" so I'm not sure it makes sense to implement Default for Transaction. Same comment for many of these. If some trait / library is forcing us to implement Default, we should at least use unreachable so we never use it, or if something does use it, for the fields we should use default, like txn_version: Default::default(), event_root_hash: Default::default(), etc.

Member
Author
@[yuunlimm](https://github.com/yuunlimm) yuunlimm [3 days ago](https://github.com/aptos-labs/aptos-indexer-processors/pull/390#discussion_r1649266202) • 
there is a usecase, where we have derive the parquet schema from the instance, and the we need to know this schema in advance. please take a look at hasParquetSchema trait in the generic_parquet_processor.rs. and we use this schema() to construct parquetHandler for each struct type. that's the only usecase of default but I am not too sure if there is any other workaround.